File: 프로그래머스/1/388351. 유연근무제/유연근무제.java
import java.util.Set;

class Solution {
    private boolean checkAttendance(int desireTime, int attendanceTime) {
        int desiredTotal = ((desireTime / 100) * 60) + (desireTime % 100);
        int attendanceTotal = ((attendanceTime / 100) * 60) + (attendanceTime % 100);
        
        return attendanceTotal - desiredTotal <= 10;
    }
    
    public int solution(int[] schedules, int[][] timelogs, int startday) {
        int answer = 0;
        int satIdx = (12 - (startday - 1)) % 7;
        int sunIdx = (satIdx + 1) % 7;
        Set<Integer> weekendIdx = Set.of(satIdx, sunIdx);
        
        for (int i = 0; i < timelogs.length; i++) {
            boolean attendedAllWeekdays = true;
            for (int j = 0; j < timelogs[i].length; j++) {
                if (weekendIdx.contains(j)) continue;
                if (!checkAttendance(schedules[i], timelogs[i][j])) {
                    attendedAllWeekdays = false;
                    break;
                }
            }
            
            if (attendedAllWeekdays) answer++;
        }
        
        return answer;
    }
}

File: 프로그래머스/1/388351. 유연근무제/유연근무제.kt
class Solution {
    private fun checkAttendance(desireTime: Int, attendanceTime: Int): Boolean {
        val desiredTotal = (desireTime / 100) * 60 + (desireTime % 100)
        val attendanceTotal = (attendanceTime / 100) * 60 + (attendanceTime % 100)

        return attendanceTotal - desiredTotal <= 10
    }
    
    fun solution(schedules: IntArray, timelogs: Array<IntArray>, startday: Int): Int {
        var answer: Int = 0
        val satIdx = (12 - (startday - 1)) % 7
        val sunIdx = (satIdx + 1) % 7
        val weekendIdx = setOf(satIdx, sunIdx)
        
        for (i in timelogs.indices) {
            val attendedAllWeekdays = timelogs[i].indices
                .filter { it !in weekendIdx }
                .all { checkAttendance(schedules[i], timelogs[i][it]) }

            if (attendedAllWeekdays) answer++
        }
        
        return answer
    }
}